Fixes Diabetes Type selector. Closes #288.

PaoloRotolo opened the pull request:
Please review.

emartynov commented:
I was reviewing but I didn't get point of the fix. What exactly is fixing the problem?

PaoloRotolo commented:
So originally there were only 2 diabetes types (1 and 2). Now we added other 2 types as well, but they were only added in the array in strings.xml. The selector in preferences had a if condition
```
if (selectedType.equals(typesArray[0])) {
   // set type 1
} else {
   // set type 2
}
```

I just added the other two types so they can be saved from preferences.

emartynov commented on app/src/main/java/org/glucosio/android/activity/PreferencesActivity.java:
```diff
@@ -214,13 +214,23 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
+
+                    if (selectedType.equals(typesArray[0])) {
                         updatedUser.setD_type(1);
                         updateDB();
```
You probably can call `updateDB()` after all if/else cases

emartynov commented:
LGTM
